fix(collect_static): LonaServer does not need `app` anymore

SmithChart opened the pull request:
This fixes a regression introduced in fe7edb65a89994ee767ea9c8f290d3ef147ff0c6.
This commit removes the `app` argument for the LonaServer(). Instead the aiohttp application is now provided inside the LonaServer (respecting the client_max_size setting.
Modules depending on the app can access is from inside the LonaServer.

This change, however, has not been reflected in the `collect-static` command line action.

Since collect static does not use the app we can simply remove the parameter.

SmithChart committed:
fix(collect_static): LonaServer does not need `app` anymore

This fixes a regression introduced in
fe7edb65a89994ee767ea9c8f290d3ef147ff0c6.
This commit removes the `app` argument for the LonaServer().
Instead the aiohttp application is now provided inside the LonaServer
(respecting the client_max_size setting.
Modules depending on the app can access is from inside the LonaServer.

This change, however, has not been reflected in the `collect-static`
command line action.

Since collect static does not use the app we can simply remove the
parameter.

Signed-off-by: Chris Fiege <cfi@pengutronix.de>

fscherf commented:
Hi @SmithChart,

Thanks for this patch, and i agree with it, but by looking at the code i think that is not enough to fix the collect-static command because i see it still uses the old client pre compiler object. Do you have time to add a patch for that too?

fscherf committed:
fix(collect-static): remove usage of old client pre compiler API

This patch fixes a regression, introduced in
7d8b5513b72d ("client: remove client pre compiler").

Before 7d8b5513b72d, the collect-static mechanism used the client pre-compiler
to get a list of all client JavaScript files to copy to the output. This API
is no longer available, because the client pre-compiler was removed.

This patch creates a new method in the static file loader, which returns all
found static files, and changes collect-static to use it.

Signed-off-by: Florian Scherf <mail@florianscherf.de>

fscherf commented:
@SmithChart: I added a second patch which fixes the second regression. Let me know if you are ok with these changes.

fscherf commented on lona/command_line/collect_static.py:
```diff
@@ -79,34 +76,8 @@ def _cp(source, destination):
-            _cp(source, destination)
-
-    # copy client
-    for source_file in server._client_pre_compiler.get_source_files():
```
The client gets served as normal static file since the pre compiler was removed

maratori reviewed (approved):
LGTM

fscherf committed:
tests: add tests for collect static

Signed-off-by: Florian Scherf <mail@florianscherf.de>

fscherf committed:
test project: frontend: add static files test view

This patch adds a test view that tests LonaView.STATIC_FILES and
AbstractNode.STATIC_FILES.

Signed-off-by: Florian Scherf <mail@florianscherf.de>

fscherf commented:
@SmithChart, @maratori:

Since the PR is still open anyway, and this won't be the last time this feature breaks, i added some tests

fscherf commented:
I think it's good enough reviewed :) @SmithChart: thank's for contributing!
